### ISSUE TITLE
Feature request: Add .net standard library support

### ISSUE THREAD

**Issue by litan1106** [skyhop/Mail#14]
Problem:
I couldn't use this library in a N-Tier architecture application.

Solution:
It would be great to have the interfaces move to a new Skyhop.Mail.Abstractions nuget package.

**Comment by synercoder** [comment]
Might be a good idea, what do you think @corstian?

**Comment by corstian** [comment]
Sounds good to me, though I'm not really familiar with an N-Tier architecture, so I'd need some guidance there.

**Comment by synercoder** [comment]
I'll pick it up somewhere in the near-future

**Comment by synercoder** [comment]
We'll talk it over on Skype somewhere this/next week. Sounds good?

**Comment by corstian** [comment]
Yep, just back from holiday so I'm good to go. Just give me a call when it fits.

**Comment by litan1106** [comment]
> Sounds good to me, though I'm not really familiar with an N-Tier architecture, so I'd need some guidance there.

Here is a great example of N-Tier real world application.

https://github.com/dotnet-architecture/eShopOnWeb

https://github.com/ardalis/CleanArchitecture

**Comment by synercoder** [comment]
Time slipped by, pandemic f*cks with your sense of time... Still assigned to me, and will try to pick it up asap.

**Comment by corstian** [comment]
@litan1106 @synercoder It took a while to get there, but there's an open pull request https://github.com/skyhop/Mail/pull/15 where I split the project into two packages;

1. Skyhop.Mail
2. Skyhop.Mail.Abstractions

Would this fit well within an n-tiered architecture and solve the current problems?

Some changes along the way;

1. All interface have been put in the Abstractions package
2. There's a new `IMailDispatcher` interface.
3. The `MailBase` class has been moved to the Abstractions package, and the BodyBuilder within has been made public get-only.

**Comment by corstian** [comment]
The 3.0.0-rc5 version has been pushed to NuGet, and contains these libraries; https://www.nuget.org/packages/Skyhop.Mail/3.0.0-rc5
